cmd/benchttp/run.go

```diff
@@ -0,0 +1,151 @@
+	config config.Global
+}
+
+// ensure cmdRun implements command
+var _ command = (*cmdRun)(nil)
+
+// execute runs the benchttp runner: it parses CLI flags, loads config
+// from config file and parsed flags, then runs the benchmark and outputs
+// it according to the config.
+func (cmd cmdRun) execute(args []string) error {
+	fieldsSet := cmd.parseArgs(args)
+
+	cfg, err := cmd.makeConfig(fieldsSet)
+	if err != nil {
+		return err
+	}
+
+	req, err := cfg.Request.Value()
+	if err != nil {
+		return err
+	}
+
+	ctx, cancel := context.WithCancel(context.Background())
+	go cmd.listenOSInterrupt(cancel)
+
+	rep, err := requester.New(cmd.requesterConfig(cfg)).Run(ctx, req)
+	if err != nil {
+		if errors.Is(err, requester.ErrCanceled) {
+			if err := cmd.handleRunInterrupt(); err != nil {
+				return err
+			}
+		} else {
+			return err
+		}
+	}
+
+	return output.New(rep, cfg).Export()
+}
+
+// parseArgs parses input args as config fields and returns
+// a slice of fields that were set by the user.
+func (cmd *cmdRun) parseArgs(args []string) []string {
+	// first arg is subcommand "run"
+	if len(args) <= 1 {
+		return []string{}
+	}
+
+	// config file path
+	cmd.flagset.StringVar(&cmd.configFile,
+		"configFile",
+		configfile.Find(cmd.defaultConfigFiles),
+		"Config file path",
+	)
+
+	// cli config
+	configflags.Set(cmd.flagset, &cmd.config)
+
+	cmd.flagset.Parse(args[1:]) //nolint:errcheck // never occurs due to flag.ExitOnError
+
+	return configflags.Which(cmd.flagset)
+}
+
+// makeConfig returns a config.Config initialized with config file
+// options if found, overridden with CLI options listed in fields
+// slice param.
+func (cmd cmdRun) makeConfig(fields []string) (cfg config.Global, err error) {
+	fileConfig, err := configfile.Parse(cmd.configFile)
+	if err != nil && !errors.Is(err, configfile.ErrFileNotFound) {
+		// config file is not mandatory, other errors are critical
+		return
+	}
+
+	mergedConfig := fileConfig.Override(cmd.config, fields...)
+
+	return mergedConfig, mergedConfig.Validate()
+}
+
+// requesterConfig returns a requester.Config generated from cfg.
+func (cmd cmdRun) requesterConfig(cfg config.Global) requester.Config {
+	return requester.Config{
+		Requests:       cfg.Runner.Requests,
+		Concurrency:    cfg.Runner.Concurrency,
+		Interval:       cfg.Runner.Interval,
+		RequestTimeout: cfg.Runner.RequestTimeout,
+		GlobalTimeout:  cfg.Runner.GlobalTimeout,
+		Silent:         cfg.Output.Silent,
+	}
+}
+
+// listenOSInterrupt listens for OS interrupt signals and calls callback.
+// It should be called in a separate goroutine from main as it blocks
+// the execution until the OS interrupt signal is received.
+func (cmd cmdRun) listenOSInterrupt(callback func()) {
+	sigC := make(chan os.Signal, 1)
+	signal.Notify(sigC, os.Interrupt)
+	<-sigC
+	callback()
+}
+
+// handleRunInterrupt handles the case when the runner is interrupted.
+func (cmd cmdRun) handleRunInterrupt() error {
```

Comment:
Nitpick: it seems it does not need to be a method of cmdRun still. Same question :)

Reply:
Well same answer as https://github.com/benchttp/runner/pull/35#discussion_r815693485: what it does is specific to the `run` command. It's so specific that it could be declared/inlined inside `cmd.Execute`, but I extracted it for better readability. However I don't see the point of extracting it one step further at the package scope 🤔